lib/min_heap.rb

```diff
@@ -14,18 +15,28 @@ def initialize
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(n)
+  # Space Complexity: O(1)
   def add(key, value = key)
-    raise NotImplementedError, "Method not implemented yet..."
+    
+    @store.push(HeapNode.new(key,value))
+    child = @store.length - 1
+    heap_up(child) if child != 0 
+
   end
 
   # This method removes and returns an element from the heap
   #   maintaining the heap structure
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(n)
+  # Space Complexity: O(1)
   def remove()
```

Comment:
The Binary tree nature of the heap and `heap_down` method mean the time complexity is O(log n) and the recursion causes a space complexity of O(log n) as well.